src/i_oalsound.c

```diff
@@ -352,9 +365,197 @@ const char **I_OAL_GetResamplerStrings(void)
+    }
+    if (!alEffectf)
+    {
+        alEffectf = FUNCTION_CAST(LPALEFFECTF, alGetProcAddress("alEffectf"));
+    }
+    if (!alAuxiliaryEffectSloti)
+    {
+        alAuxiliaryEffectSloti = FUNCTION_CAST(LPALAUXILIARYEFFECTSLOTI, alGetProcAddress("alAuxiliaryEffectSloti"));
+    }
+    if (!alGenFilters)
+    {
+        alGenFilters = FUNCTION_CAST(LPALGENFILTERS, alGetProcAddress("alGenFilters"));
+    }
+    if (!alDeleteFilters)
+    {
+        alDeleteFilters = FUNCTION_CAST(LPALDELETEFILTERS, alGetProcAddress("alDeleteFilters"));
+    }
+    if (!alIsFilter)
+    {
+        alIsFilter = FUNCTION_CAST(LPALISFILTER, alGetProcAddress("alIsFilter"));
+    }
+
+    if (!alGenAuxiliaryEffectSlots ||
+        !alDeleteAuxiliaryEffectSlots ||
+        !alIsAuxiliaryEffectSlot ||
+        !alGenEffects ||
+        !alDeleteEffects ||
+        !alIsEffect ||
+        !alEffecti ||
+        !alEffectf ||
+        !alAuxiliaryEffectSloti ||
+        !alGenFilters ||
+        !alDeleteFilters ||
+        !alIsFilter)
+    {
+        return;
+    }
+
+    if (alIsAuxiliaryEffectSlot(uiEffectSlot))
+    {
+        for (int i = 0; i < MAX_CHANNELS; i++)
+        {
+            alSource3i(oal->sources[i], AL_AUXILIARY_SEND_FILTER, AL_EFFECTSLOT_NULL, 0, AL_FILTER_NULL);
+        }
+
+        alDeleteAuxiliaryEffectSlots(1, &uiEffectSlot);
+    }
+
+    if (alIsEffect(uiEffect))
+    {
+        alDeleteEffects(1, &uiEffect);
+    }
+
+    if (alIsFilter(uiFilter))
+    {
+        alDeleteFilters(1, &uiFilter);
+    }
+
+    if (snd_equalizer != EQ_PRESET_OFF)
+    {
+        if (!alIsAuxiliaryEffectSlot(uiEffectSlot))
+        {
+            alGenAuxiliaryEffectSlots(1, &uiEffectSlot);
+        }
+
+        if (!alIsEffect(uiEffect))
+        {
+            alGenEffects(1, &uiEffect);
+        }
+
+        if (!alIsFilter(uiFilter))
+        {
+            alGenFilters(1, &uiFilter);
+        }
+
+        alEffecti(uiEffect, AL_EFFECT_TYPE, AL_EFFECT_EQUALIZER);
+
+        // Gains vary from 0.126 up to 7.943, which means from -18dB attenuation
+        // up to +18dB amplification, i.e. 20*log10(gain).
+
+        #define GAIN_PCT(pct) ((ALfloat)BETWEEN(0.126f, 7.943f, pct/100.f))
+
+        alEffectf(uiEffect, AL_EQUALIZER_LOW_GAIN, GAIN_PCT(snd_eq_low_gain));
+        alEffectf(uiEffect, AL_EQUALIZER_MID1_GAIN, GAIN_PCT(snd_eq_mid1_gain));
+        alEffectf(uiEffect, AL_EQUALIZER_MID2_GAIN, GAIN_PCT(snd_eq_mid2_gain));
+        alEffectf(uiEffect, AL_EQUALIZER_HIGH_GAIN, GAIN_PCT(snd_eq_high_gain));
+
+        alEffectf(uiEffect, AL_EQUALIZER_LOW_CUTOFF, (ALfloat)snd_eq_low_cutoff);
+        alEffectf(uiEffect, AL_EQUALIZER_HIGH_CUTOFF, (ALfloat)snd_eq_high_cutoff);
+
+        alAuxiliaryEffectSloti(uiEffectSlot, AL_EFFECTSLOT_EFFECT, uiEffect);
+
+        for (int i = 0; i < MAX_CHANNELS; i++)
+        {
+            alSource3i(oal->sources[i], AL_AUXILIARY_SEND_FILTER, uiEffectSlot, 0, uiFilter);
```

Comment:
You may have noticed that even with "flat" settings, the volume is louder when the EQ is enabled. That's because the dry gain needs to be muted otherwise both the original "direct/dry" and modified "aux/wet" sound paths play at the same time. It's not documented very well, but refer to [this diagram](https://openal-soft.org/misc-downloads/Effects%20Extension%20Guide.pdf#%5B%7B%22num%22%3A937%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C592%2Cnull%5D) (direct filter and aux filter), and [this table](https://openal-soft.org/misc-downloads/Effects%20Extension%20Guide.pdf#%5B%7B%22num%22%3A1924%2C%22gen%22%3A0%7D%2C%7B%22name%22%3A%22XYZ%22%7D%2C0%2C123%2Cnull%5D) (Attenuation on Direct path).

You can apply this patch: [diff.txt](https://github.com/user-attachments/files/15567177/diff.txt)

Reply:
Yes, that bothered my. But I thought it was just me and my hearing. :grin:

Reply:
No problem, I wonder if going back to the dB scale for the slider would be fine now.

Reply:
I'm not sure if we really want sliders in the menu, but I agree that the dB scale is much more intuitive.

Reply:
Oops, I meant scale in the config, not slider.